src/Tokenizer.php

```diff
@@ -848,222 +831,101 @@ private function makeRegexFromList(array $values, bool $sorted = false): string
-            } else {
-                // Non-quoted variable name
-                preg_match('/\G[@:][\w.$]+/', $string, $matches, 0, $offset);
-                if ($matches) {
-                    $value = $matches[0];
-                }
-            }
-
-            if ($value !== null) {
-                return new Token($type, $value);
-            }
-        }
-
-        // Number (decimal, binary, or hex)
-        if (
-            preg_match(
-                $this->nextTokenRegexNumber,
-                $string,
-                $matches,
-                0,
-                $offset,
-            )
-        ) {
-            return new Token(Token::TOKEN_TYPE_NUMBER, $matches[0]);
-        }
-
-        // Boundary Character (punctuation and symbols)
-        if (preg_match($this->nextTokenRegexBoundaryCharacter, $string, $matches, 0, $offset)) {
-            return new Token(Token::TOKEN_TYPE_BOUNDARY, $matches[0]);
-        }
-
-        // A reserved word cannot be preceded by a '.'
-        // this makes it so in "mytable.from", "from" is not considered a reserved word
-        if ($previous === null || $previous->value() !== '.') {
-            // Top Level Reserved Word
-            if (
-                preg_match(
-                    $this->nextTokenRegexReservedToplevel,
-                    $upper,
-                    $matches,
-                    0,
-                    $offset,
-                )
-            ) {
-                return new Token(
-                    Token::TOKEN_TYPE_RESERVED_TOPLEVEL,
-                    substr($string, $offset, strlen($matches[0])),
-                );
-            }
+        $tokens = [];
+        $offset = 0;
 
-            // Newline Reserved Word
-            if (
-                preg_match(
-                    $this->nextTokenRegexReservedNewline,
-                    $upper,
-                    $matches,
-                    0,
-                    $offset,
-                )
-            ) {
-                return new Token(
-                    Token::TOKEN_TYPE_RESERVED_NEWLINE,
-                    substr($string, $offset, strlen($matches[0])),
-                );
-            }
+        while ($offset < strlen($string)) {
+            // Get the next token and the token type
+            preg_match($tokenizeRegex, $upper, $matches, 0, $offset);
+            assert(($matches[0] ?? '') !== '');
 
-            // Other Reserved Word
-            if (
-                preg_match(
-                    $this->nextTokenRegexReserved,
-                    $upper,
-                    $matches,
-                    0,
-                    $offset,
-                )
-            ) {
-                return new Token(
-                    Token::TOKEN_TYPE_RESERVED,
-                    substr($string, $offset, strlen($matches[0])),
-                );
+            while (is_int($lastMatchesKey = array_key_last($matches))) {
+                array_pop($matches);
```

Comment:
If removing integer keys from the array is what we want, can't we do this more efficiently with `array_filter()`?

Reply:
We cannot probably do better, we are interested only in the last non-int key. `array_filter` would filter the whole array.